framework/finalizer.go

```diff
@@ -117,6 +117,14 @@ func createAddFinalizerPatch(obj interface{}, operatorName string) (patch []patc
 		Path:  "/metadata/finalizers/-",
 	}
 	patch = append(patch, addPatch)
+
+	testResourceVersionPatch := patchSpec{
+		Op:    "test",
```

Comment:
So in general when there is a race. And I believe there always is because of the cache. The framework needs to wait for next reconciliation loop to proceed?